hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -1129,6 +1129,14 @@ public void refreshNodes() throws IOException {
     rpcClient.invokeConcurrent(nss, method, true, true);
   }
 
+  @Override
+  public void refreshTopology() throws IOException {
+    // Router not support this operation, because this maybe refresh multi namespaces
+    String methodName = RouterRpcServer.getMethodName();
+    throw new UnsupportedOperationException(
+        "Operation \"" + methodName + "\" is not supported");
+  }
+
```

Comment:
in case of router it should delegate the request to all namespaces, that is how all other commands of similar nature work